src/ledger/LedgerDelta.cpp

```diff
@@ -417,13 +417,8 @@ LedgerDelta::markMeters(Application& app) const
 }
 
 void
-LedgerDelta::checkAgainstDatabase(Application& app) const
+LedgerDelta::checkAgainstDatabase(Database& db) const
 {
-    if (!app.getConfig().PARANOID_MODE)
```

Comment:
This check is moved to: https://github.com/stellar/stellar-core/pull/1161/files#diff-26d755120733c7309bb85f71f8fcd2c9R207

The whole problem was that PARANOID_MODE was enabled in all tests config, but was ignored as implementation of applyCheck in tests did something a bit different than normal closeLedger. I've modified applyCheck to work exactly as closeLedger (by calling it) when possible (when Transaction is valid), so we won't encounter that kind of bug again.

Also applyCheck requires some additional checks of LedgerDelta (starts with verify "modified accounts invariants"), so I've added parameter to closeLedger to perform these checks. It was also a good place to extract checkAgainstDatabase() from that place and move check for PARANOID_MODE up in the ownership chain. This removed dependency on app.getConfig() from LedgerDelta and made checkAgainstDatabase() do what it says in the name (before that this methods should be named checkAgainstDatabaseIfParanoidMode()).